bpf/protocol_http2.h

```diff
@@ -24,6 +25,30 @@ struct {
     __uint(pinning, BEYLA_PIN_INTERNAL);
 } ongoing_http2_grpc SEC(".maps");
 
+typedef struct grpc_frames_ctx {
+    http2_grpc_request_t prev_info;
+    u8 has_prev_info;
+
+    int pos; //FIXME should be size_t equivalent
```

Comment:
This FIXME will be addressed separately - there are too many other places to check and tweak.

Reply:
(copying this here as I commented on the wrong box - sorry about the noise)

With regards to the int usage, my biggest concern is that we have a signed int here - in most APIs (except from a few relics like libc functions such as read(), which will rely on negative values for errors, and anything else for "byte count"), we will encounter size_t, which tends to be the same as uintptr_t because it needs to be able to address the memory space. We do invariably end up mixing these ints up with pointers and alike, which are unsigned by default, so IMHO it's just best to eventually refactor them to become unsigned (and simplify underflow/overflow cases).

Using a smaller unsigned int here is potentially fine, but we have to be careful to not make it appear as an lvalue in an operation involving pointers as rvalues, as it can cause an overflow (because pointers will be 64-bit). It becomes a trade-off between saving a few bytes (or a lot depending on how many of those we have) versus less cognitive load of treating everything as size_t.